File: app/src/main/java/com/tangxb/demo/fortest/api/AndroidResultApi.java
package com.tangxb.demo.fortest.api;


import com.tangxb.demo.fortest.bean.CommonBean;
import com.tangxb.demo.fortest.bean.ResultBean;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by Taxngb on 2017/11/15.
 */
public interface AndroidResultApi {
    @GET("data/Android/{pageSize}/{page}")
    Observable<CommonBean<List<ResultBean>>> getData(@Path("pageSize") int pageSize, @Path("page") int page);
}

File: app/src/main/java/com/tangxb/demo/api/AndroidResultApi.kt
package com.tangxb.demo.api

import com.tangxb.demo.bean.CommonBean
import com.tangxb.demo.bean.ResultBean
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Created by Taxngb on 2017/11/14.
 */
interface AndroidResultApi {
    @GET("data/Android/{pageSize}/{page}")
    fun getData(@Path("pageSize") pageSize: Int, @Path("page") page: Int): Observable<CommonBean<List<ResultBean>>>
}